lib/generator.js

```diff
@@ -113,26 +113,42 @@ function buildOperations(spec) {
   }
 
   function buildOperationObject(pathName, methodName, method) {
+    const parameters = (method.parameters || []);
     return Object.assign({
       path: pathName,
       method: methodName,
-      queryParams: method.parameters
-        .filter(param => param.in === 'query')
-        .map(param => {
-          if (param['x-openapi-v3-schema-ref']) {
-            param.model = getModelName(param['x-openapi-v3-schema-ref']);
-            delete param['x-openapi-v3-schema-ref'];
+      queryParams: parameters
+      .filter(param => param.in === 'query')
+      .map(param => {
+        if (param['x-openapi-v3-schema-ref']) {
+          param.model = getModelName(param['x-openapi-v3-schema-ref']);
+          delete param['x-openapi-v3-schema-ref'];
+        }
+        return param;
+      }),
+      pathParams: parameters.filter(param => param.in === 'path'),
+      consumes: (() => {
+        const content = _.get(method, 'requestBody.content');
+        return content ? Object.keys(content) : [];
+      })(),
+      produces: (() => {
+        if (method.responses) {
+          const mediaTypes = new Set();
+          for (const [statusCode, response] of Object.entries(method.responses)) {
+            if (response.content) {
+              Object.keys(response.content)
+                .forEach(mediaType => mediaTypes.add(mediaType));
+            }
           }
-          return param;
-        }),
-      pathParams: method.parameters.filter(param => param.in === 'path')
+          return Array.from(mediaTypes);
+        }
+        return [];
+      })(),
```

Comment:
Rather than change `operations` to reflect the OAS3 shape, I decided to keep the current contract to minimize the impact on consumers.